packages/core/src/package-utils/bundle.ts

```diff
@@ -43,26 +43,25 @@ export const createBundle = async (
         );
 
         // internal package resolved by plugins/vite/internal-package-resolution.ts
-        if (srcPath.startsWith('../')) return;
-
-        assert(
-          !srcPath.startsWith('./'),
-          `relative path ${srcPath} should not start with './'`,
-        );
+        if (srcPath.startsWith('../')) {
+          logger.debug(`Internal package ${srcPath}`);
+          return;
+        }
 
         if (
           isVanillaFile(id) ||
           getModuleInfo(id)?.importers.some(isVanillaFile)
         ) {
-          return `${stylesDir}/${srcPath}`;
+          return normalizePath(`${stylesDir}/${srcPath}`);
         }
 
         if (
-          typeof packageJson.sideEffects !== 'boolean' &&
+          typeof packageJson.sideEffects !== 'undefined' &&
           moduleHasSideEffects(srcPath, packageJson.sideEffects) &&
```

Comment:
Doesn't this conflict with the implementation of `moduleHasSideEffects`?
It would return `true` if given `undefined`, but now if it's `undefined` it doesn't even get to go through `moduleHasSideEffects`.

Reply:
This is the correct behaviour. Without a `"sideEffects"` flag we don't make any decisions about files with side-effects i.e. we don't move them to `dist/side-effects` and we don't update the flag (because it doesn't exist). For us, the `"sideEffects"` flag is important for those two reasons only and we leave everything else to the bundlers.

Also, the function is copied as-is [from webpack](https://github.com/webpack/webpack/blob/v5.72.1/lib/optimize/SideEffectsFlagPlugin.js#L322) (and I didn't want to change anything), that's why the `undefined` check is outside.